coral/src/app/layout/header/ProfileDropdown.tsx

```diff
@@ -32,20 +32,22 @@ const menuItems: MenuItem[] = [
 
 const LOGOUT_KEY = "logout";
 function ProfileDropdown() {
-  const toast = useToast();
+  const [toast, dismiss] = useToastContext();
   function navigateToAngular(path: string) {
     window.location.assign(`${window.origin}${path}`);
   }
 
   function onDropdownClick(actionKey: string | number) {
     if (actionKey === LOGOUT_KEY) {
       toast({
+        id: "logout",
         message: "You are being logged out of Klaw...",
         position: "bottom-left",
         variant: "default",
-        duration: 30 * 1000,
       });
       logoutUser().catch((error) => {
+        // dismiss toast in case logout fails
```

Comment:
Could you move the dismissing of the toast down into the `error.status !== 401` condition? 

I checked the UI and the toast closes a bit soon, so there's a short moment where the toast is gone but nothing happens yet, because the checks for the error status are done in the background. That could look a bit misleading for users. 

So we should close the toast only in case there's an error, then it won't close too soon. 

(Since we're redirecting the user in case it's the correct status, the component ins unmounted anyway and we don't need to take care of closing the toast)